autokeras/image_supervised.py

```diff
@@ -186,30 +186,24 @@ def fit(self, x_train=None, y_train=None, time_limit=None):
         Args:
             x_train: A numpy.ndarray instance containing the training data.
             y_train: A numpy.ndarray instance containing the label of the training data.
+            x_test: A numpy.ndarray instance containing the testing data
+            y_test: A numpy.ndarray instance containing the label of the testing data.
             time_limit: The time limit for the search in seconds.
         """
-        if y_train is None:
-            y_train = []
-        if x_train is None:
-            x_train = []
-
         x_train = np.array(x_train)
         y_train = np.array(y_train).flatten()
-
         _validate(x_train, y_train)
-
         y_train = self.transform_y(y_train)
-
+        if x_test is None or y_test is None:
+            # Divide training data into training and testing data.
+            x_train, x_test, y_train, y_test = train_test_split(x_train, y_train,
+                                                                test_size=min(Constant.VALIDATION_SET_SIZE,
+                                                                              int(len(y_train) * 0.2)),
+                                                                random_state=42)
         # Transform x_train
         if self.data_transformer is None:
             self.data_transformer = DataTransformer(x_train, augment=self.augment)
```

Comment:
The data transformer should be initialized before the split. The transformer init calculates the mean and standard deviation from the data. It should use the whole training set.